docs/nima/builder.adoc

```diff
@@ -0,0 +1,210 @@
+   <plugins>
+       <plugin>
+           <groupId>org.apache.maven.plugins</groupId>
+           <artifactId>maven-compiler-plugin</artifactId>
+           <version>3.10.1</version>
+           <configuration>
+               <annotationProcessorPaths>
+                   <path>
+                       <groupId>io.helidon.builder</groupId>
+                       <artifactId>helidon-builder-processor</artifactId>
+                       <version>4.0.0-ALPHA4</version>
+                   </path>
+               </annotationProcessorPaths>
+           </configuration>
+       </plugin>
+   </plugins>
+</build>
+
+
+----
+
+== Usage
+
+Helidon Builder can be used in the following scenarios:
+
+=== Coding and Testing
+
+The following examples illustrate how to use Helidon Builder with simple test subject classes. To create your test subject classes, create an example with two interface types: `NameAndDescription` and `NameAndDescriptionAndComments` that extends from the first, each having the Builder annotation:
+
+.NameAndDescription
+[source, java]
+
+----
+@Builder(requireLibraryDependencies = false)
+public interface NameAndDescription {
+   String name();
+   Optional<String> description();
+}
+
+----
+
+.NameAndDescriptionAndComments
+[source, java]
+
+----
+@Builder(requireLibraryDependencies = false)
+public interface NameAndDescriptionAndComments extends NameAndDescription {
+   @Singular
+   List<String> comments();
+}
+
+----
+
+
+NOTE: The two classes above use `requireLibraryDependencies=false` which allows you to use Maven's provided dependency for the Helidon Builder libraries. This also makes it possible to code-generate the supporting validation directly into your builder- generated types, thereby not requiring any Helidon library support in your runtime.
+
+
+.Build Your Project
+[source, java]
+
+----
+maven clean compile
+----
+
+.Write your main()
+[source, java]
+
+----
+public class Main {
+   public static void main(String[] args) {
+       NameAndDescriptionAndComments target =
+               DefaultNameAndDescriptionAndComments.builder()
+                       .name("world")
+                       .addComment("hello")
+                       .build();
+       System.out.println("Hello " + target);
+   }
+}
+----
+
+.Output
+[source, java]
+
+----
+
+Hello NameAndDescriptionAndComments(name=world, description=Optional.empty,
+comments=[hello])
+
+----
+
+
+
+=== Auto-Generated Code
+
+The Helidon Builder's annotation processor generates source code whenever it processes an annotation that it recognizes - in this case the @Builder annotation. You must apply the annotation processors in your build in order to generate the supporting sources before they are consumed in the main() from this example.
+
+
+
+
+== Configuration
+
+//I wasn't sure how to position Builder Config. It seems like it fits in with Builder but perhaps we want to keep it separate? Also, are we using the term Pico Builder Config or something else?
```

Comment:
ConfigBean Builder is an extensibility feature over the base Builder to allow integration of the generated builder types to the Helidon Configuration subsystem.

Reply:
Oh, I see you have it below.